lib/model/store.dart

```diff
@@ -451,6 +453,7 @@ class LivePerAccountStore extends PerAccountStore {
       initialSnapshot: initialSnapshot,
     );
     store.poll();
+    store.registerNotificationToken();
```

Comment:
> As implemented, it looks like its only dependencies are […]

I guess it rightly has another dependency: some means of tracking when the listener it adds on `NotificationService.instance.token` needs to be removed. That's a convenient role for `LivePerAccountStore` to fill…hmm. I see the "TODO call removeListener on [dispose]" in `registerNotificationToken`.

Reply:
> Does the token-registration call want to go earlier than this, which comes after `/register`?

Hmm, good thought. I think I basically just copied this aspect from zulip-mobile.




> I guess it rightly has another dependency: some means of tracking when the listener it adds on `NotificationService.instance.token` needs to be removed. That's a convenient role for `LivePerAccountStore` to fill…hmm. I see the "TODO call removeListener on [dispose]" in `registerNotificationToken`.

Yeah. But that's really just because on dispose, we may be about to make a new `LivePerAccountStore` for the same account — so it's a way of preventing duplication. If the listener lived somewhere else that had a different lifetime, then we'd want to cancel it based on that thing's lifetime instead.

I guess there's the complication (which we'll eventually implement) that the user's API key could become invalid, or the user or realm get deactivated. In that case we'll want to tear down whatever has that listener, along with the `LivePerAccountStore` and/or whatever's doing the polling for events.





> …But actually, does the job of `registerNotificationToken` really need to wait until we have a `LivePerAccountStore`? (Does it need to be a method on `LivePerAccountStore`?)

Yeah, I don't totally love having it here. It's here because it seems parallel to the event-polling loop `poll`; and because this is the most obvious place for the code to live if it's going to be initiated after we get the register response, i.e. from `load`.